pymc/sampling.py

```diff
@@ -466,11 +458,37 @@ def sample(
 
     draws += tune
 
+    auto_nuts_init = True
+    if step is not None:
+        if isinstance(step, CompoundStep):
+            for method in step.methods:
+                if isinstance(method, NUTS):
+                    auto_nuts_init = False
+        elif isinstance(step, NUTS):
+            auto_nuts_init = False
+
     initial_points = None
     step = assign_step_methods(model, step, methods=pm.STEP_METHODS, step_kwargs=kwargs)
 
     if isinstance(step, list):
         step = CompoundStep(step)
+    elif isinstance(step, NUTS) and auto_nuts_init:
+        if "nuts" in kwargs:
+            nuts_kwargs = kwargs.pop("nuts")
+            [kwargs.setdefault(k, v) for k, v in nuts_kwargs.items()]
+        _log.info("Auto-assigning NUTS sampler...")
+        initial_points, step = init_nuts(
+            init=init,
+            chains=chains,
+            n_init=n_init,
+            model=model,
+            seeds=random_seed,
+            progressbar=progressbar,
+            jitter_max_retries=jitter_max_retries,
+            tune=tune,
+            initvals=initvals,
+            **kwargs,
+        )
```

Comment:
I'm not confident about this. As I commented before, the `try/catch` block caught not-implemented gradients, __which we're probably not testing__.

Also I'm sceptical about the pattern of having `isinstance(step, NUTS)` and `if "nuts" in kwargs`. It reads like a hack to account for some specific cases, and it's not as general as the previous implementation.

You're the ones who've been looking into this much more than me, so feel free to overrule this.

Reply:
The assign_step_methods checks for gradients as well. If that does not do a good enough job of testing if nuts is appropriate than it should show up as a bug and be fixed there

Reply:
Okay, you're right. Also we're testing it: https://app.codecov.io/gh/pymc-devs/pymc/blob/18346acc797c32ccfa2d61ebc8f368ef03bc216c/pymc/sampling.py